app/models/location.rb

```diff
@@ -1,5 +1,37 @@
+require 'geokit'
+
 class Location < ActiveRecord::Base
   belongs_to :user
 
   validates :user, presence: true
+
+  after_save :reindex_dishes
+
+  def reindex_dishes
+    user.dishes.each { |dish| dish.reindex }
+  end
+
+  # only update latlng when moved distance > 0.5 mile
+  # this is to reduce the cost whenever location is changed, have to reindex dishes (ElasticSearch)
+  def update_latlng(new_lat, new_lng)
+    new_lat = new_lat.try(:to_f) if !new_lat.blank?
+    new_lng = new_lng.try(:to_f) if !new_lng.blank?
+
+    # calculate moved distance
+    dist = distance_from new_lat, new_lng
+
+    # only when location change is noticeable
+    if dist and dist > 0.5 # 1 mile
+      update_attributes(lat: new_lat, lng: new_lng) 
+    end
+  end
+
+  def distance_from(another_lat, another_lng)
+    if lat and lng and another_lat and another_lng
```

Comment:
Consider checking arguments first before doing anything else in this method. Then  `return`-ing early or `raise`-ing an `ArgumentError` if the arguments are unexpected.

Also favour using `&&` instead of `and` in `if` conditions, as this is more conventional. Other Rubyists reading your code are more likely to understand it.